usr/src/cmd/tip/hunt.c

```diff
@@ -72,10 +72,10 @@ hunt(char *name)
 			t.c_cflag |= XCLUDE|HUPCL;
 			(void) ioctl(FD, TCSETSF, &t);
 			(void) signal(SIGALRM, f);
-			return ((int)cp);
+			return ((intptr_t)cp);
```

Comment:
The function here is still returning `int`.  I can't tell if you meant to fix that, or are relying on it implicitly truncating again.

Reply:
Yeah sorry trying to work around some function that is looking crazy, fixed.

Reply:
From reading the call sites, I think this should actually be
```suggestion
			return (cp != NULL);
```

and the same at the end of the function, let's see if github will let me comment on that one